bauhaus/workflows/secondary/hqMetrics.py

```diff
@@ -16,66 +17,135 @@ def conditionTableType():
 
     def generate(self, pflow, ct):
         outputDict = {}
-        pflow.bundleScript("R/hqrf_pbbamr.R")
+        rscript = "R/hqrf_pbbamr.R"
+        pflow.bundleScript(rscript)
+        controls = "fasta/600bp_Control_c2.fasta"
+        pflow.bundleScript(controls)
+
+        # R script merges  nohq_aligned reads with HQR regions
         pflow.genRuleOnce("hqrf_pbbamr",
-                          "$grid Rscript --verbose R/hqrf_pbbamr.R $origbam xscraps $in $out")
+                          "$grid Rscript --verbose " + rscript + " $subreadsbam $scrapsbam $in $out")
 
+        # strips out HQR region information (forming zmw reads) and re-does adapter finding on the zmw reads
         pflow.genRuleOnce("bam2bam_nohq",
                           "$grid bam2bam -j $ncpus -b $ncpus --fullHQ --adapters $adapters -o $outPrefix $in")
 
+        # converts BAZ file to subreads.bam, doing adapter finding and control filtering
+        pflow.genRuleOnce("baz2bam",
+                          "$grid baz2bam -j $ncpus -b $ncpus --minSnr=0 --metadata $metadata --adapters $adapters --controls $controls -o $outPrefix $in")
+
+        # runs basecaller on trace file, outputting BAZ file
+        pflow.genRuleOnce("t2b",
+                          "$grid stdbuf -o 0 $basecaller --internal --input $in --output $out --logoutput $log")
+
+        # creates a subreadset dataset from bam files
         pflow.genRuleOnce("subreadset_create",
                           "$grid dataset create $out $in")
 
         for condition in ct.conditions:
             with pflow.context("condition", condition):
                 reference = ct.reference(condition)
-                if ct.inputType == InputType.SubreadSet:
-                    inputs = ct.inputs(condition)
 
+                inputs = ct.inputs(condition)
+
+                # the workflow can start with TraceH5File or SubreadSet
+                # TODO : add BAZFile as a starting point
+                if ct.inputType == InputType.TraceH5File:
                     if len(inputs) != 1 :
                         raise NotImplementedError, "condition must be 1:1 with inputs"
 
-                    with pflow.context("movieName", movieName(inputs[0])):
-                         # 1. remove HQ regions
+                    trch5 = inputs[0] + ""
+                    (d,b) = os.path.split(trch5)
+                    movieName_ = b.replace(".trc.h5","")
+
+                    # TODO: find meta data and/or adapters in a better way (not assume a naming convention)
+                    metadata = os.path.join(d,"." + movieName_ + ".run.metadata.xml")
```

Comment:
I should mention that I had planned to break the flow into 3 separate stages (trc.h5 -> baz -> bam -> HQRF metrics), but for now I just put them together so we could run this now before refactoring.